lang/src/org/partiql/lang/eval/like/Pattern.kt

```diff
@@ -0,0 +1,89 @@
+package org.partiql.lang.eval.like
+
+import java.util.regex.Pattern
+
+private const val ANY_MANY = '%'.toInt()
+private const val ANY_ONE = '_'.toInt()
+
+/**
+ * Translates a SQL-style `LIKE` pattern to a regular expression.
+ *
+ * @param likePattern A `LIKE` match pattern (i.e. a string where '%' means '.*?' and '_' means '.').
+ * @param escapeChar The escape character for the `LIKE` pattern.
+ *
+ * @return a [Pattern] which is a regular expression corresponding to the specified `LIKE` pattern.
+ */
+internal fun parsePattern(likePattern: String, escapeChar: Int?): Pattern {
```

Comment:
Nit: For better ergonomics, reckon we could have the common escape chars as an Enum? (E.g. `'\\'.toInt()`)

Reply:
Also, does `escapeChar` need to be nullable?

Reply:
According to the spec:

> The optional third argument is a character string containing exactly one character, known as the "escape character", for use when a percent or underscore is required in the pattern without its special meaning.

The [PostGres docs say](https://www.postgresql.org/docs/current/functions-matching.html#FUNCTIONS-LIKE):

> According to the SQL standard, omitting ESCAPE means there is no escape character (rather than defaulting to a backslash), and a zero-length ESCAPE value is disallowed. PostgreSQL's behavior in this regard is therefore slightly nonstandard.

There are a lot of existing test cases that pass `null` here: https://github.com/partiql/partiql-lang-kotlin/blob/12051302413f8d3c3c87aafa7551c0c93d8d0648/lang/test/org/partiql/lang/eval/like/PatternTests.kt#L15-L148

I didn't change the signature from the removed PatternParts: https://github.com/partiql/partiql-lang-kotlin/blob/9da23cfb07672580d4b56b671ad18f044a850e60/lang/src/org/partiql/lang/eval/like/PatternPart.kt#L27